idl/github.com/uber/cadence/shared.thrift

```diff
@@ -24,6 +24,14 @@ exception BadRequestError {
   1: required string message
 }
 
+exception BlobEncodingError {
```

Comment:
Can we just use BadRequestError instead of adding these new error types?  BadRequestError is already treated as non-retryable throughout the codebase.